pkgs/top-level/aliases.nix

```diff
@@ -351,6 +351,7 @@ mapAliases ({
   dart_dev = throw "Non-stable versions of Dart have been removed"; # Added 2020-01-15
   dart_old = throw "Non-stable versions of Dart have been removed"; # Added 2020-01-15
   dart_stable = dart; # Added 2020-01-15
+  darwin.builder = throw "'darwin.builder' has been renamed to 'darwin.linux-builder' and requires a migration: https://nixos.org/manual/nixpkgs/unstable/#sec-darwin-builder";
```

Comment:
@roberth any suggestions on how to write this alias? This fails with:

```
error:
       … while evaluating the attribute 'darwin.overrideScope'

         at /nix/store/2l7lgxsidza81llpq5imhxf3a7cdk1zl-source/pkgs/stdenv/darwin/default.nix:651:9:

          650|
          651|         darwin = super.darwin.overrideScope (_: _: {
             |         ^
          652|           inherit (darwin) dyld ICU Libsystem Csu libiconv rewrite-tbd;

       … while evaluating a branch condition

         at /nix/store/2l7lgxsidza81llpq5imhxf3a7cdk1zl-source/pkgs/top-level/aliases.nix:33:5:

           32|   removeDistribute = alias: with lib;
           33|     if isDerivation alias then
             |     ^
           34|       dontDistribute alias

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Alias darwin is still in all-packages.nix
```

Reply:
I think `aliases.nix` is only for top level aliases, but `builder` is inside the `darwin` attribute, and `darwin` is by itself not an alias.

You could still support `config.allowAliases` by manually making your attribute optional based on that bool.

Reply:
Updated :+1: